### PR TITLE
test-suite-kotlin

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.uri.UriBuilder
+import java.util.Optional
+import java.util.UUID
+import javax.validation.Valid
+
+@Controller("/customer")
+internal class CustomerController(private val repository: CustomerRepository) {
+    @Post
+    fun save(@Body customerSave: @Valid CustomerSave): HttpResponse<*> {
```
Comment (timyates) [inline test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt:20]: Is putting `@Valid` before the type idiomatic?

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    override fun findByFirstName(firstName: @NotBlank String): Collection<Customer> {
+        return customersByFirstName(
+            data
+                .map { obj: Data -> obj.getCustomers() }
+                .orElseGet { emptyList() }, firstName
+        )
+    }
+
+    private val data: Optional<Data>
+        private get() {
```
Comment (timyates) [inline test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt:50]: nit: Don't think private is required

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        XThreads.executeSynchronized {
+            data.ifPresent { data: Data ->
+                data.add(customer)
+            }
+            embeddedStorageManager.storeAll()
+        }
+    }
+
+    @NonNull
+    override fun findById(id: @NotBlank String): Optional<Customer> {
```
Comment (timyates) [inline test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt:25]: Not sure Optional use in kotlin is ideomatic, should be `Customer?` instead?

**Review by timyates — APPROVED** [review]
Just a few nit-pics

The question about annotation position in method definitions is for my education 😉

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt (hunk 1 above)
Comment (sdelamo) [reply test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt:19]: If I change from: 

```kotlin
@Post
fun save(@Body @Valid customerSave: CustomerSave): HttpResponse<*> {
```

to: 

```kotlin
@Post
    fun save(@Body @Valid customerSave: CustomerSave): HttpResponse<*> {
``` 


I get: 


```
CustomerController.java:17: error: Method defines AOP advice but is declared final. Change the method to be non-final in order for AOP advice to be applied.
    public final io.micronaut.http.HttpResponse<?> save(@org.jetbrains.annotations.NotNull()
                                                   ^

```

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt (hunk 2 above)
Comment (sdelamo) [reply test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt]: resolved via : https://github.com/micronaut-projects/micronaut-microstream/pull/11/commits/796ef2a6f92eaaeb824ad5836f7945246a9947cb

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt (hunk 3 above)
Comment (sdelamo) [reply test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt]: I have removed the `Optional` from Kotlin code: 

https://github.com/micronaut-projects/micronaut-microstream/pull/11/commits/3bbc97ae5d6c92f0008db4d5c1e0efa5c5470423

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt (hunk 1 above)
Comment (timyates) [reply test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt:19]: Weird...  and the validation works on the save method?

**Review by timyates — APPROVED** [review]

**Pull request merged**